server/lib/abstract_feed_stream.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Base class for feed streams.
+ **/
+
+import { PassThrough } from 'stream';
+
+export default class AbstractFeedStream extends PassThrough {
+
+  constructor() {
+    super({
+      decodeStrings: true,
+      objectMode: true,
+    });
+  }
+
+  /**
+   * The abstract function which is responsible for handling feed requests to
+   * specific services. Needs to be implemented in child classes.
+   **/
+  streamFeed() {
+    throw new Error('The "streamFeed" method must be implemented!');
```

Comment:
Maybe you could move this string to some constant? We would avoid looking for all the places where it was used, if we want to change the message in the future.